### ISSUE TITLE
Folder named test is missing from installed dependency

### ISSUE THREAD

**Issue by claudiopetrini** [npm/npm#19200]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
npm isn't installing a folder named `test` that's contained in a dependency.

#### How can the CLI team reproduce the problem?
With npm `5.4.2`
```
git clone https://github.com/claudiopetrini/depfromtesting.git
cd depfromtesting
npm install
```
The resulting `node_modules` will contain the `testing` folder without the folder named `test`, why?
The `testing` [repo](https://github.com/claudiopetrini/testing) contains that folder.

Shell output:
```
$ git clone https://github.com/claudiopetrini/depfromtesting.git
Cloning into 'depfromtesting'...
remote: Counting objects: 3, done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 3 (delta 0), reused 3 (delta 0), pack-reused 0
Unpacking objects: 100% (3/3), done.
$ cd depfromtesting/
$ npm install
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN depfromtesting@1.0.0 No description

added 1 package in 0.746s
$ cd node_modules/testing/
$ ls -la
total 16
drwxr-xr-x  5 user  group   160 20 Nov 12:43 .
drwxr-xr-x  3 user  group    96 20 Nov 12:43 ..
drwxr-xr-x  3 user  group    96 20 Nov 12:43 development
-rw-r--r--  1 user  group   136 15 Nov 17:34 index.js
-rw-r--r--  1 user  group  1441 20 Nov 12:43 package.json
```

### supporting information:

 - `npm -v` prints: `5.4.2`
 - `node -v` prints: `v8.8.1`
 - `npm config get registry` prints: `https://registry.npmjs.org/`
 - Windows, OS X/macOS, or Linux?: `macOS v10.13.1`
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by kenany** [comment]
I cannot reproduce this with `npm@5.5.1`:

```console
$ git clone https://github.com/claudiopetrini/depfromtesting.git
$ cd depfromtesting/
$ npm i
$ ls node_modules/testing/
development  index.js  package.json  test
```

**Comment by Butterfly** [comment]
I played along from a lurking position to learn a little more tonight. Mine succeeded to the point the sought after `test` is there. I'm on Debian/Buster (Testing)/Linux.. Node v8.9.1, NPM 5.5.1. Most of note right now is that I have *nothing else* installed beyond `nodejs` that brought `npm` along with..

@claudiopetrini  What directory were you in when you ran into that? I just created any ol' directory under my normal ~/ user. I can take the few seconds this took to run it again somewhere else just to see if that makes a difference, possibly about maybe permissions (again) or something. :smile:

**Comment by claudiopetrini** [comment]
On a fresh AWS Instance with Amazon Linux AMI 2017.09.1 (`ami-6057e21a`) it's possible to reproduce the issue with node 8.8.1 and npm 5.4.2.

```
sudo yum install -y git
curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.6/install.sh | bash
```
Exit and ssh again:
```
nvm install 8.8.1   
git clone https://github.com/claudiopetrini/depfromtesting.git
cd depfromtesting/
npm i
```

This doesn't happen when using node 8.9.1 and npm 5.5.1.

**Comment by kenany** [comment]
Sounds like it's been fixed then?

**Comment by claudiopetrini** [comment]
Yes, it can be considered fixed.

**Comment by claudiopetrini** [comment]
For reference: reading the `npm` [changelog](https://github.com/npm/npm/releases) I've found this:
```
npm-packlist@1.1.9 Fix a bug where test directories would always be excluded from published modules.
```
Fixed in `v5.5.0`
